app/selectors/server.js

```diff
@@ -0,0 +1,8 @@
+import { createSelector } from 'reselect';
+
+const getServer = state => state.share.server || state.server.server;
+
+export const getServerSelector = createSelector(
```

Comment:
This is a cool improvement, but it should be done on a separate PR changing all server usage.
Otherwise, we're just adding fragmentation.